src/CTA.Rules.Update/CodeReplacer.cs

```diff
@@ -44,23 +53,57 @@ public CodeReplacer(List<SourceFileBuildResult> sourceFileBuildResults, ProjectC
+                            {
+                                File.WriteAllText(sourceFileBuildResult.SourceFileFullPath, result);
+                            }
+
+                            var processedActions = ValidateActions(oneRewriter.allExecutedActions, result);
+                            processedActions = AddActionsWithoutExecutions(currentFileActions, oneRewriter.allExecutedActions);
+
+                            if (!actionsPerProject.TryAdd(sourceFileBuildResult.SourceFileFullPath, processedActions))
+                            {
+                                throw new FilePortingException(sourceFileBuildResult.SourceFilePath, new Exception("File already exists in collection"));
+                            }
+                        }                       
+                        else
                         {
-                            throw new FilePortingException(sourceFileBuildResult.SourceFilePath, new Exception("File already exists in collection"));
+                            if (currentFileActions != null)
+                            {
+                                currentFileActions.NodeTokens.ForEach(nodetoken =>
+                                {
+                                    ActionsRewriter oneRewriter = new ActionsRewriter(sourceFileBuildResult.SemanticModel, sourceFileBuildResult.PrePortSemanticModel, sourceFileBuildResult.SyntaxGenerator, currentFileActions.FilePath, nodetoken.AllActions);
+                                    
+                                    var result = root.NormalizeWhitespace().ToFullString();
+                                    //If true, line endings and spaces need to be normalized:
+                                    //if (result != root.ToFullString())
+                                    //{
+                                    //    File.WriteAllText(sourceFileBuildResult.SourceFileFullPath, result);
+                                    //}
+
+                                    var newRoot = oneRewriter.Visit(root);
+                                    var allChanges = newRoot.SyntaxTree.GetChanges(root.SyntaxTree);
+                                    
+                                    foreach (var textChange in allChanges)
+                                    {
+                                        var fileLinePositionSpan = root.SyntaxTree.GetMappedLineSpan(textChange.Span);
+                                        var newTextChange = new TextChange() { FileLinePositionSpan = fileLinePositionSpan, NewText = textChange.NewText };
+                                        if (!nodetoken.TextChanges.Contains(newTextChange))
+                                        {
+                                            nodetoken.TextChanges.Add(newTextChange);
+                                        }
+                                    }
+                                });
+                            }
```

Comment:
Each branch in this `if...else` block is quite large. What do you think about moving the logic for each branch into descriptive private functions to help with readability? For example:
```csharp
if (_projectConfiguration.PortCode)
{
    ApplyCodeActionsAndPersist( ... );
}
else
{
    // assumes TextChanges are used to preview changes
    ApplyCodeActionsAndPreview( ... ); 
}
```